### PR TITLE
[RW-7074][risk=no] e2e Rework open tab function

### PR DESCRIPTION
**Pull request by aweng98**
Test failure:
`openDatasetsSubtab()` function fails indeterminately in `dataset-snowman-menu.spec`. e.g. [here](https://app.circleci.com/pipelines/github/all-of-us/workbench/11990/workflows/a1ee5e9c-05aa-4ac9-a73c-d6ea3b1fa714/jobs/151491/parallel-runs/3).

UI cause:
`aria-selected` attribute is not a reliable information anymore for determining wether a tab is open or not. Data tab sometimes has `aria-selected=false` when tab is open. It's a minor UI bug recently introduced. I don't know who or what PR has caused it.

Screen Shot 2021-07-19 at 11 31 09 AM

Proposed fix:
A workaround in test functions to avoid repeated failures: Don't rely on `aria-selected` attribute.

New `openDataSubtab()` function will determine if subtab exists. If found it, click it. If not found, find and click Data tab first then find the subtab.

### CODE REVIEW

File: e2e/tests/datasets/dataset-snowman-menu-actions.spec.ts [hunk 1]
```diff
@@ -102,16 +102,13 @@ describe('Datasets card snowman menu actions', () => {
...
-    await dataPage.openDatasetsSubtab();
-
     const newDatasetExists = await resourceCard.cardExists(newDatasetName, ResourceCard.Dataset);
     expect(newDatasetExists).toBe(true);
 
     const oldDatasetExists = await resourceCard.cardExists(datasetName, ResourceCard.Dataset);
     expect(oldDatasetExists).toBe(false);
 
-    await dataPage.deleteResource(newDatasetName, ResourceCard.Dataset);
+    await resourceCard.delete(newDatasetName, ResourceCard.Dataset);
```
Comment (aweng98) [inline e2e/tests/datasets/dataset-snowman-menu-actions.spec.ts:111]: Less hassle to write code that deletes resource card.

**Comment by aweng98** [comment]
Hi Will, I wonder if you can help with review? Ticket was filed by you last week. Thank you.

**Review by dolbeew — APPROVED** [review]
LGTM

**Comment by aweng98** [comment]
Thank you very much.

**Pull request merged**